cmd/kubeadm/app/phases/certs/certs.go

```diff
@@ -117,8 +117,80 @@ func CreateAPIServerKubeletClientCertAndKeyFiles(cfg *kubeadmapi.MasterConfigura
 		cfg.CertificatesDir,
 		kubeadmconstants.APIServerKubeletClientCertAndKeyBaseName,
 		caCert,
-		apiClientCert,
-		apiClientKey,
+		apiKubeletClientCert,
+		apiKubeletClientKey,
+	)
+}
+
+// CreateEtcdServerCertAndKeyFiles create a new certificate and key file for etcd.
+// If the etcd serving certificate and key file already exist in the target folder, they are used only if evaluated equal; otherwise an error is returned.
+// It assumes the cluster CA certificate and key file exist in the CertificatesDir
+func CreateEtcdServerCertAndKeyFiles(cfg *kubeadmapi.MasterConfiguration) error {
+
+	caCert, caKey, err := loadCertificateAuthority(cfg.CertificatesDir, kubeadmconstants.CACertAndKeyBaseName)
```

Comment:
I think unless we can guarantee that a k8s client cert does not grant access directly to etcd and an etcd client cert does not grant access to the api server, then defaulting to a separate CA for etcd/k8s is probably a good idea here.

Reply:
Thanks @detiber!
After today's banter about CN-based Authentication with etcd v3.3 being difficult to implement in the short-term, an immediate path we can take is to implement the separate etcd CA. This may be the simplest, secure option.

@jbeda also mentioned using Etcd RBAC to achieve similar results without the flag from the v3.3 PR: https://github.com/coreos/etcd/pull/8616.
I don't know anything about this right now.

Reply:
Looks like you can make etcd authN a user based off of a cert's CN with `--client-cert-auth=true` and `etcdctl auth enable`
https://coreos.com/etcd/docs/latest/op-guide/authentication.html#using-tls-common-name

The new `--peer-cert-allowed-cn` implies that this doesn't affect Peer Certificates though.
Maybe it's because that's before the cluster is bootstrapped? 🤔

Reply:
> unless we can guarantee that a k8s client cert does not grant access directly to etcd

^ we've started discussing this

> and an etcd client cert does not grant access to the api server

^ we haven't really considered this other half yet